include/wil/com_apartment_variable.h

```diff
@@ -217,11 +217,13 @@ namespace details
         // get current value or throw if no value has been set
         std::any& get_existing()
         {
-            if (auto any = get_if())
+            auto any = get_if();
+            if (!any)
             {
-                return *any;
+                THROW_HR(E_NOT_SET);

```

Comment:
Note that changes like this are a consequence of defining `RESULT_NORETURN` to nothing. I debated trying to change it so that only the failfast functions are marked with `RESULT_NORETURN` since those are the only ones that we detour, however this is simple enough